### PR TITLE
AMP-44814 Android Kotlin example

### PR DESCRIPTION
**Pull request by falconandy**

### CODE REVIEW

File: kotlin/android/AmpliApp/app/build.gradle [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    implementation 'androidx.core:core-ktx:1.7.0'
+    implementation 'androidx.appcompat:appcompat:1.4.0'
+    implementation 'com.google.android.material:material:1.4.0'
+    implementation 'androidx.constraintlayout:constraintlayout:2.1.2'
+    testImplementation 'junit:junit:4.+'
+    androidTestImplementation 'androidx.test.ext:junit:1.1.3'
+    androidTestImplementation 'androidx.test.espresso:espresso-core:3.4.0'
+
+    implementation 'com.amplitude:android-sdk:2.34.1'
+    implementation 'com.squareup.okhttp3:okhttp:4.2.2'
```
Comment (justin-fiedler) [inline kotlin/android/AmpliApp/app/build.gradle:45]: I had to use `com.squareup.okhttp3:okhttp:4.9.3` does `4.2.2` work for you? https://github.com/amplitude/Amplitude-Android/issues/309

File: kotlin/android/AmpliApp/app/src/androidTest/java/com/example/ampliapp/ExampleInstrumentedTest.kt [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+import org.junit.Assert.*
+
+/**
+ * Instrumented test, which will execute on an Android device.
+ *
+ * See [testing documentation](http://d.android.com/tools/testing).
+ */
+@RunWith(AndroidJUnit4::class)
+class ExampleInstrumentedTest {
```
Comment (justin-fiedler) [inline kotlin/android/AmpliApp/app/src/androidTest/java/com/example/ampliapp/ExampleInstrumentedTest.kt:17]: Can you please add basic sanity tests here like in Node, Browser & iOS.

**Review by justin-fiedler — CHANGES_REQUESTED** [review]
@falconandy this is looking awesome!

File: kotlin/android/AmpliApp/app/src/main/java/com/example/ampliapp/App.kt [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+        val amplitudeApiKey: String? = null // Set your Amplitude API key
+        ampli.load(
+            LoadOptions(
+                androidContext = this,
+                environment = Ampli.Environment.DEVELOPMENT,
+                client = LoadClientOptions(apiKey = amplitudeApiKey)
+            )
+        )
+    }
```
Comment (justin-fiedler) [inline kotlin/android/AmpliApp/app/src/main/java/com/example/ampliapp/App.kt:18]: Can you please add (commented out) examples of other Ampli usages like this - https://github.com/amplitude/ampli-examples/blob/main/browser/typescript/react-app/src/App.tsx#L17-L52

File: README.md [hunk 4]
```diff
@@ -11,4 +11,7 @@ Example apps using the Ampli SDK
 * [node/javascript](node/javascript)
 
 ## Node TypeScript
-* [node/typescript](node/typescript)
\ No newline at end of file
+* [node/typescript](node/typescript)
+
+## Android Kotlin
+* [kotlin/android/AmpliApp](kotlin/android/AmpliApp)
```
[the remark this answers is not in the document]
Comment (falconandy) [reply README.md]: Renamed

File: kotlin/android/AmpliApp/app/src/main/java/com/example/ampliapp/App.kt [hunk 5]
```diff
@@ -0,0 +1,19 @@
+package com.example.ampliapp
+
+import android.app.Application
+import com.amplitude.ampli.*
+
+class App : Application() {
+    override fun onCreate() {
+        super.onCreate()
+
+        val amplitudeApiKey: String? = null // Set your Amplitude API key
```
[the remark this answers is not in the document]
Comment (falconandy) [reply kotlin/android/AmpliApp/app/src/main/java/com/example/ampliapp/App.kt]: Done

File: kotlin/android/AmpliApp/app/src/androidTest/java/com/example/ampliapp/ExampleInstrumentedTest.kt (hunk 2 above)
Comment (falconandy) [reply kotlin/android/AmpliApp/app/src/androidTest/java/com/example/ampliapp/ExampleInstrumentedTest.kt]: Added

File: kotlin/android/AmpliApp/app/src/main/java/com/example/ampliapp/App.kt (hunk 3 above)
Comment (falconandy) [reply kotlin/android/AmpliApp/app/src/main/java/com/example/ampliapp/App.kt]: Added. I've changed `load` signature from `load(options)` to `load(appContext, options?)` to simplify default usage

File: kotlin/android/AmpliApp/app/build.gradle (hunk 1 above)
Comment (falconandy) [reply kotlin/android/AmpliApp/app/build.gradle]: I see similar warnings/errors in log but events are sent successfully - I can see them in Amplitude dashboards

File: android/kotlin/AmpliApp/README.md [hunk 6]
```diff
@@ -0,0 +1,30 @@
+# Ampli Android SDK (Kotlin)
+An example app using the Ampli Android SDK
+
+# Usage
+
+### Setup the project
+You will need to do the following before running the app.
+1. Create new file `src/main/res/values/secrets.xml` with your Amplitude API key:
+```xml
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (justin-fiedler) [inline android/kotlin/AmpliApp/README.md:10]: Can you please add this file explicitly to git as described here, then gitignore

File: android/kotlin/AmpliApp/app/build.gradle [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+    implementation 'androidx.core:core-ktx:1.7.0'
+    implementation 'androidx.appcompat:appcompat:1.4.0'
+    implementation 'com.google.android.material:material:1.4.0'
+    implementation 'androidx.constraintlayout:constraintlayout:2.1.2'
+    testImplementation 'junit:junit:4.+'
+    androidTestImplementation 'androidx.test.ext:junit:1.1.3'
+    androidTestImplementation 'androidx.test.espresso:espresso-core:3.4.0'
+
+    implementation 'com.amplitude:android-sdk:2.35.2'
+    implementation 'com.squareup.okhttp3:okhttp:4.2.2'
```
Comment (justin-fiedler) [inline android/kotlin/AmpliApp/app/build.gradle:45]: Please update to `4.9.3` and add comment with link to this issue https://github.com/amplitude/Amplitude-Android/issues/309

File: android/kotlin/AmpliApp/README.md [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+<resources>
+    <string name="AMPLITUDE_API_KEY">your-amplitide-api-key</string>
+</resources>
+```
+
```
Comment (justin-fiedler) [inline android/kotlin/AmpliApp/README.md:15]: Can you please add a `local.properties.example` with `#sdk.dir=/path/to/Android/sdk` and reference it here

File: android/kotlin/AmpliApp/app/src/main/java/com/example/ampliapp/App.kt [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+        //     LoadOptions(
+        //         client = LoadClientOptions(instance = instance)
+        //     )
+        // )
+
+
+        val amplitudeApiKey = getString(R.string.AMPLITUDE_API_KEY)
+        ampli.load(this,
+            LoadOptions(
+                environment = Ampli.Environment.DEVELOPMENT,
```
Comment (justin-fiedler) [inline android/kotlin/AmpliApp/app/src/main/java/com/example/ampliapp/App.kt:56]: environment is ignored here, is there a reason to add this versus uncommenting the example above?

File: android/kotlin/AmpliApp/app/src/main/java/com/example/ampliapp/App.kt [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+        //     )
+        // )
+
+        /**
+         * OR Use an existing Amplitude instance
+         * requires "import com.amplitude.api.Amplitude"
+         */
+        // val instance = Amplitude.getInstance()
+        // instance.initialize(this, AMPLITUDE_API_KEY)
+        // ampli.load(this,
```
Comment (justin-fiedler) [inline android/kotlin/AmpliApp/app/src/main/java/com/example/ampliapp/App.kt:46]: To keep the example more concise could we use this format for all load examples
```
ampli.load(this, LoadOptions(
    client = LoadClientOptions(apiKey = AMPLITUDE_API_KEY)
))
```

File: android/kotlin/AmpliApp/app/src/test/java/com/example/ampliapp/ExampleUnitTest.kt [hunk 11]
```diff
@@ -0,0 +1,17 @@
...
+import org.junit.Assert.*
+
+/**
+ * Example local unit test, which will execute on the development machine (host).
+ *
+ * See [testing documentation](http://d.android.com/tools/testing).
+ */
+class ExampleUnitTest {
+    @Test
+    fun addition_isCorrect() {
```
Comment (justin-fiedler) [inline android/kotlin/AmpliApp/app/src/test/java/com/example/ampliapp/ExampleUnitTest.kt:14]: Please add basic unit tests (load, identify, group, track, event method)

**Review by justin-fiedler — CHANGES_REQUESTED** [review]
Very close. See comments. Please also resolve merge conflict.

File: android/kotlin/AmpliApp/app/src/test/java/com/example/ampliapp/ExampleUnitTest.kt (hunk 11 above)
Comment (falconandy) [reply android/kotlin/AmpliApp/app/src/test/java/com/example/ampliapp/ExampleUnitTest.kt:14]: @justin-fiedler Do you mean we need new unit tests in addition to https://github.com/amplitude/ampli-examples/blob/AMP-44814-android-kotlin-example/android/kotlin/AmpliApp/app/src/androidTest/java/com/example/ampliapp/AmpliTest.kt?

File: android/kotlin/AmpliApp/app/build.gradle (hunk 7 above)
Comment (falconandy) [reply android/kotlin/AmpliApp/app/build.gradle]: Done

File: android/kotlin/AmpliApp/README.md (hunk 8 above)
Comment (falconandy) [reply android/kotlin/AmpliApp/README.md:20]: Done. Not sure how to write text correctly

File: android/kotlin/AmpliApp/app/src/main/java/com/example/ampliapp/App.kt (hunk 9 above)
Comment (falconandy) [reply android/kotlin/AmpliApp/app/src/main/java/com/example/ampliapp/App.kt]: Deleted

File: android/kotlin/AmpliApp/app/src/main/java/com/example/ampliapp/App.kt (hunk 10 above)
Comment (falconandy) [reply android/kotlin/AmpliApp/app/src/main/java/com/example/ampliapp/App.kt]: Done

File: kotlin/android/AmpliApp/app/src/androidTest/java/com/example/ampliapp/ExampleInstrumentedTest.kt (hunk 2 above)
Comment (justin-fiedler) [reply kotlin/android/AmpliApp/app/src/androidTest/java/com/example/ampliapp/ExampleInstrumentedTest.kt]: Thank you!

File: android/kotlin/AmpliApp/app/src/test/java/com/example/ampliapp/ExampleUnitTest.kt (hunk 11 above)
Comment (justin-fiedler) [reply android/kotlin/AmpliApp/app/src/test/java/com/example/ampliapp/ExampleUnitTest.kt:14]: Sorry, current tests in androidTest are good.  Was confused why this file was empty.

**Review by justin-fiedler — APPROVED** [review]
Awesome work!

**Pull request merged**